spec/unit/recipes/setup_spec.rb

```diff
@@ -0,0 +1,125 @@
+#
+# Cookbook Name:: setup
+# Spec:: default
+#
+# Copyright 2015 The Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+
+describe 'setup' do
+  let(:unit_types) do
+    %w(service socket device mount automount swap target path timer slice)
+  end
+
+  cached(:chef_run) do
+    runner = ChefSpec::SoloRunner.new(step_into: unit_types)
+    runner.converge(described_recipe)
+  end
+
+  it '#service' do
+    expect(chef_run).to create_systemd_service('test-unit').with(
+      description: 'Test Service',
+      documentation: 'man:true(1)',
+      aliases: %w( testing-unit.service testd.service ),
```

Comment:
sorry, the interface for this actually changed to enforce aliasing to the same unit type by using the resource uni_type to add the unit-type extension for this directive, so instead of testing-unit.service, user can just specify testing-unit, and resource will alias to same unit type as itself.